### PR TITLE
fix: Make dry_run optional per docs

### LINKED ISSUE #34563: DryRun is not optional for patch task instance
## Summary

According to the [REST api docs](https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/patch_mapped_task_instance).
you can patch a task instance state. When you hit the api without sending a
"dry_run" variable, you get a KeyError (This is from a server running version 2.5.3):

```
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/airflow/.local/lib/python3.10/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/home/airflow/.local/lib/python3.10/site-packages/connexion/decorators/decorator.py", line 68, in wrapper
    response = function(request)
  File "/home/airflow/.local/lib/python3.10/site-packages/connexion/decorators/uri_parsing.py", line 149, in wrapper
    response = function(request)
  File "/home/airflow/.local/lib/python3.10/site-packages/connexion/decorators/validation.py", line 196, in wrapper
    response = function(request)
  File "/home/airflow/.local/lib/python3.10/site-packages/connexion/decorators/validation.py", line 399, in wrapper
    return function(request)
  File "/home/airflow/.local/lib/python3.10/site-packages/connexion/decorators/response.py", line 112, in wrapper
    response = function(request)
  File "/home/airflow/.local/lib/python3.10/site-packages/connexion/decorators/parameter.py", line 120, in wrapper
    return function(**kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/api_connexion/security.py", line 51, in decorated
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 75, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/api_connexion/endpoints/task_instance_endpoint.py", line 594, in patch_task_instance
    if not data["dry_run"]:
KeyError: 'dry_run'
```

The API docs state that dry_run is not required and that it is defaulted to false.

This can be reproduced in `main` with the tests by commenting out line 1699 in
[test_task_instance_endpoint.py](https://github.com/apache/airflow/blob/5b0ce3db4d36e2a7f20a78903daf538bbde5e38a/tests/api_connexion/endpoints/test_task_instance_endpoint.py#L1695-L1709)

### PR DESCRIPTION
**Pull request by Calder-Ty**
This fixes #34563. It makes SetSingleTaskInstanceStateFormSchema load dry_run as False by default.
This matches what is declared in the api docs

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: tests/api_connexion/endpoints/test_task_instance_endpoint.py [hunk 1]
```diff
@@ -210,7 +210,8 @@ def test_should_respond_200(self, username, session):
         # This prevents issue when users upgrade to 2.0+
         # from 1.10.x
         # https://github.com/apache/airflow/issues/14421
-        session.query(TaskInstance).update({TaskInstance.operator: None}, synchronize_session="fetch")
+        session.query(TaskInstance).update(
+            {TaskInstance.operator: None}, synchronize_session="fetch")
```
Comment (hussein-awala) [inline tests/api_connexion/endpoints/test_task_instance_endpoint.py]: Try to avoid unnecessary and unrelated changes
```suggestion
        session.query(TaskInstance).update({TaskInstance.operator: None}, synchronize_session="fetch")
```

File: tests/api_connexion/endpoints/test_task_instance_endpoint.py [hunk 2]
```diff
@@ -312,7 +313,8 @@ def test_should_respond_200_with_task_state_in_deferred(self, session):
         }
 
     def test_should_respond_200_with_task_state_in_removed(self, session):
-        self.create_task_instances(session, task_instances=[{"state": State.REMOVED}], update_extras=True)
+        self.create_task_instances(session, task_instances=[
+                                   {"state": State.REMOVED}], update_extras=True)
```
Comment (hussein-awala) [inline tests/api_connexion/endpoints/test_task_instance_endpoint.py]:
```suggestion
        self.create_task_instances(session, task_instances=[{"state": State.REMOVED}], update_extras=True)
```

File: tests/api_connexion/endpoints/test_task_instance_endpoint.py [hunk 3]
```diff
@@ -667,7 +669,8 @@ def test_should_respond_200_for_dag_id_filter(self, session):
         )
 
         assert response.status_code == 200
-        count = session.query(TaskInstance).filter(TaskInstance.dag_id == "example_python_operator").count()
+        count = session.query(TaskInstance).filter(
```
Comment (hussein-awala) [inline tests/api_connexion/endpoints/test_task_instance_endpoint.py]:
```suggestion
        count = session.query(TaskInstance).filter(TaskInstance.dag_id == "example_python_operator").count()
```

**Review by hussein-awala — COMMENTED** [review]
IMHO it's a problem in the doc, dry_run should be `True` by default.

File: tests/api_connexion/endpoints/test_task_instance_endpoint.py (hunk 1 above)
Comment (Calder-Ty) [reply tests/api_connexion/endpoints/test_task_instance_endpoint.py]: Easy fix, will turn off my formatter

**Comment by Calder-Ty** [comment]
> IMHO it's a problem in the doc, dry_run should be `True` by default.

I don't care about what the default is, however that change would technically be a breaking change. Albeit, no one can be depending on the declared functionality from the docs as it is broken.

**Comment by Calder-Ty** [comment]
I did not mean to close this. But will open a new PR.
